starfish/core/imagestack/imagestack.py

```diff
@@ -1152,7 +1153,8 @@ def max_proj(self, *dims: Axes) -> "ImageStack":
         max_projection = self._data.max([dim.value for dim in dims])
         max_projection = max_projection.expand_dims(tuple(dim.value for dim in dims))
         max_projection = max_projection.transpose(*self.xarray.dims)
-        max_proj_stack = self.from_numpy(max_projection.values)
+        physical_coords = self.xarray.coords
+        max_proj_stack = self.from_numpy(max_projection.values, coordinates=physical_coords)
```

Comment:
the coordinates passed in here should be shaped like the array passed in here.  the z averaging should happen here.